.dotnet/src/Custom/Chat/ChatResponseFormat.cs

```diff
@@ -1,45 +1,71 @@
-using OpenAI.Models;
+using OpenAI.Internal;
+using System;
+using System.ClientModel.Primitives;
+using System.ComponentModel;
 
 namespace OpenAI.Chat;
 
-/// <summary>
-/// Represents a requested <c>response_format</c> for the model to use, enabling "JSON mode" for guaranteed valid output.
-/// </summary>
-/// <remarks>
-/// <b>Important</b>: when using JSON mode, the model <b><u>must</u></b> also be instructed to produce JSON via a
-/// <c>system</c> or <c>user</c> message.
-/// <para>
-/// Without this paired, message-based accompaniment, the model may generate an unending stream of whitespace until the
-/// generation reaches the token limit, resulting in a long-running and seemingly "stuck" request.
-/// </para>
-/// <para>
-/// Also note that the message content may be partially cut off if <c>finish_reason</c> is <c>length</c>, which
-/// indicates that the generation exceeded <c>max_tokens</c> or the conversation exceeded the max context length for
-/// the model.
-/// </para>
-/// </remarks>
-[CodeGenModel("CreateChatCompletionRequestResponseFormat")]
-public partial class ChatResponseFormat
+[CodeGenModel("ChatResponseFormat")]
+public abstract partial class ChatResponseFormat
 {
-    // CUSTOM: Made internal.
+    public static ChatResponseFormat Text { get; } = new InternalChatResponseFormatText();
+    public static ChatResponseFormat JsonObject { get; } = new InternalChatResponseFormatJsonObject();
 
-    /// <summary> Must be one of `text` or `json_object`. </summary>
-    [CodeGenMember("Type")]
-    internal InternalCreateChatCompletionRequestResponseFormatType? Type { get; set; }
-
-    // CUSTOM: Made internal.
-    /// <summary> Initializes a new instance of <see cref="ChatResponseFormat"/>. </summary>
-    internal ChatResponseFormat()
+    public static ChatResponseFormat CreateTextFormat() => new InternalChatResponseFormatText();
+    public static ChatResponseFormat CreateJsonObjectFormat() => new InternalChatResponseFormatJsonObject();
+    public static ChatResponseFormat CreateJsonSchemaFormat(
+        string name,
+        BinaryData jsonSchema,
+        string description = null,
+        bool? requireStrictJsonSchemaMatch = null)
```

Comment:
What do you think about simply `requireStrictMatch` ?

Reply:
I think it would be ideal if the name matched (or at a minimum resembled) the name of the equivalent property in `ChatTool`, which is currently called `StrictParameterSchemaEnabled`. So calling it `strictSchemaEnabled` is probably better.

Ideally this extrapolates to Assistants too.

Reply:
Makes sense. Changed across the board to `strictSchemaEnabled`. The more verbose version was primarily just to make it super clear that it applied to the related parameter, but I agree that the more consistent name is better -- and "schema" is more than enough to figure it out.